libs/src/index.js

```diff
@@ -76,18 +76,18 @@ class AudiusLibs {
   /**
    * Configures an internal web3 to use (via Hedgehog)
    * @param {string} registryAddress
-   * @param {Array} allProviderUrls web3 provider endpoints
+   * @param {string | object | Array<string>} providers web3 provider endpoint(s)
    */
-  static configInternalWeb3 (registryAddress, allProviderUrls) {
-    if (typeof allProviderUrls === 'string') {
-      allProviderUrls = [allProviderUrls]
+  static configInternalWeb3 (registryAddress, providers) {
+    if (typeof providers === 'string') {
+      providers = [providers]
```

Comment:
should we split by ',' if it's a comma separated str? i feel like it's actually broken now cause if someone passes in a comma separated

Reply:
we split by ',' in the caller of this method (in dapp). want to keep this consistent w/ the web3manager

Reply:
so in other words, the arg options here are
'https://provider.co'
['https://provider.co', 'https://provider2.co']

Reply:
sorry forgot to respond here. looking at this again i see one issue. typeof array will also return "object" and if you wrap that in array brackets you'll be creating a nested array. also how can this be object? 

as far as the input type, youre right from an api perspective we should define it as such and ideally people use it the correct way. but my gut take is because we put more responsibility on the consumer of this to format this, there will be a bug one day because split comma separated env vars, as they're always passed into a service or client that way, will be directly passed into this list without preprocessing. i feel like we can protect against all of that by changing this case to
```
if (typeof providers !== "string" || !Array.isArray(providers)) throw new Error("invalid providers list")

if (typeof providers === "string") providers = providers.split(',')
```
and split gives you array conversion for free. 

if you feel strongly about not splitting here that's fine, i just think it's such easy defense against potential future bugs it. also that comma separated bug could still happen with TS 🙂

Reply:
Fair enough :) Updated.